test/rekt/features/broker/feature.go

```diff
@@ -37,6 +42,151 @@ import (
+		eventshub.StartReceiver),
+	)
+	f.Setup("install sink2", eventshub.Install(sink2, eventshub.StartReceiver))
+	f.Setup("install sink3", eventshub.Install(sink3, eventshub.StartReceiver))
+
+	// filter1 filters the original events
+	filter1 := eventingv1.TriggerFilterAttributes{
+		"type":   eventType,
+		"source": eventSource,
+	}
+	// filter2 filters all events
+	filter2 := eventingv1.TriggerFilterAttributes{
+		"type": eventingv1.TriggerAnyFilter,
+	}
+	// filter3 filters events after transformation
+	filter3 := eventingv1.TriggerFilterAttributes{
+		"type":   transformedEventType,
+		"source": transformedEventSource,
+	}
+
+	// Install the trigger1 point to Broker and transform the original events to new events
+	f.Setup("install trigger1", trigger.Install(
+		trigger1,
+		brokerName,
+		trigger.WithFilter(filter1),
+		trigger.WithSubscriber(svc.AsKReference(sink1), ""),
+	))
+	f.Setup("trigger1 goes ready", trigger.IsReady(trigger1))
+	// Install the trigger2 point to Broker to filter all the events
+	f.Setup("install trigger2", trigger.Install(
+		trigger2,
+		brokerName,
+		trigger.WithFilter(filter2),
+		trigger.WithSubscriber(svc.AsKReference(sink2), ""),
+	))
+	f.Setup("trigger2 goes ready", trigger.IsReady(trigger2))
+
+	// Install the channel and corresponding subscription point to sink3
+	channelName := feature.MakeRandomK8sName("channel")
+	f.Setup("install channel", channel.Install(channelName,
+		channel.WithTemplate(),
+	))
+	sub := feature.MakeRandomK8sName("subscription")
+	f.Setup("install subscription", subscription.Install(sub,
+		subscription.WithChannel(channel.AsRef(channelName)),
+		createSubscriberFn(svc.AsKReference(sink3), ""),
+	))
+	f.Setup("subscription is ready", subscription.IsReady(sub))
+	f.Setup("channel is ready", channel.IsReady(channelName))
+
+	// Install the trigger3 point to Broker to filter the events after transformation point to channel
+	f.Setup("install trigger3", trigger.Install(
+		trigger3,
+		brokerName,
+		trigger.WithFilter(filter3),
+		trigger.WithSubscriber(channel.AsRef(channelName), ""),
+	))
+	f.Setup("trigger3 goes ready", trigger.IsReady(trigger3))
+
+	f.Requirement("install source", eventshub.Install(
+		source,
+		eventshub.StartSenderToResource(broker.GVR(), brokerName),
+		eventshub.InputEvent(eventToSend),
+	))
+
+	eventMatcher := eventasssert.MatchEvent(
+		test.HasSource(eventSource),
+		test.HasType(eventType),
+		test.HasData([]byte(eventBody)),
+	)
+	transformEventMatcher := eventasssert.MatchEvent(
+		test.HasSource(transformedEventSource),
+		test.HasType(transformedEventType),
+		test.HasData([]byte(transformedBody)),
+	)
+
+	f.Stable("(Trigger2 point to) sink2 has all the events").
+		Must("delivers original events",
+			eventasssert.OnStore(sink2).Match(eventMatcher).AtLeast(1)).
+		Must("delivers transformation events",
+			eventasssert.OnStore(sink2).Match(transformEventMatcher).AtLeast(1))
```

Comment:
As a sanity check can we also verify that `sink1` has got the original event? with:
```
eventasssert.OnStore(sink1).Match(eventMatcher).AtLeast(1)
```

Reply:
In this way, if something is wrong on the assertions of `sink2` and `sink3` related to the original event we can get a better error message

Reply:
Got it